### ISSUE TITLE
request of SparseApplyFtrlOp running on gpu

### ISSUE THREAD

**Issue by sjtusmartboy** [tensorflow/tensorflow#32170]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (or github SHA if from source):  1.13.1

Currently, I notice that `SparseApplyFtrlOp` only supports running on cpu platform. Seeing from here:
https://github.com/tensorflow/tensorflow/blob/d5c6687d9919c562bea2a01a6e1be1756bfaab33/tensorflow/core/kernels/training_ops.cc#L2448

However, in some circumstances, there's great advantage of supporting this  `SparseApplyFtrlOp` in time performance. So is there any plan to support this op on gpu, and if not, what's the reason? Thanks

**Comment by sjtusmartboy** [comment]
@jvishnuvardhan @gadagashwini Any update? Thanks

**Comment by sjtusmartboy** [comment]
@jvishnuvardhan @gadagashwini Any update? Thanks

**Comment by sjtusmartboy** [comment]
@aaroey hi, is there any update?

**Comment by aaroey** [comment]
@reedwm could you help to take a look? Thanks.

**Comment by reedwm** [comment]
@tanzhenyu, I see you added the tf.keras Ftrl optimizer. Are there any plans to support GPUs with sparse gradients?

**Comment by tanzhenyu** [comment]
There are plans to support GPU. However specifically ftrl might not be the prioritized item in the plan

**Comment by reedwm** [comment]
Ok, I'll assign to you for now

**Comment by sjtusmartboy** [comment]
@tanzhenyu hi, can you be more specific, about what kind of gpu optimizer will be supported, sparse optimizer or dense optimizer?

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
